File: examples/typescript/claims/validate-professional-claim-submission.ts
/**
 * Check a professional claim without submitting it.
 *
 *   STEDI_API_KEY=<api-key> tsx validate-professional-claim-submission.ts
 *
 * Runs the same payer checks, claim edits, and X12 mapping as
 * CreateProfessionalClaimSubmission, but persists and delivers nothing — so you can
 * validate a claim while a user is still filling out a form. A response with no errors
 * means the claim would be accepted.
 */

import { readFile } from "node:fs/promises";

import { Stedi, type ValidateProfessionalClaimSubmissionInput } from "@stedi/sdk";

const FIXTURE = `${import.meta.dirname}/fixtures/test-claim.json`;

/** Reads the sample claim. Nothing is persisted, so no per-run stamping is needed. */
async function loadClaim(): Promise<ValidateProfessionalClaimSubmissionInput> {
  const claim = JSON.parse(
    await readFile(FIXTURE, "utf8"),
  ) as ValidateProfessionalClaimSubmissionInput;

  if (claim.payer?.id !== "STEDITEST") {
    throw new Error("input must bill the Stedi test payer.");
  }

  return claim;
}

const apiKey = process.env["STEDI_API_KEY"];
if (!apiKey) {
  throw new Error("STEDI_API_KEY is not set");
}

const client = new Stedi({ apiKey: { apiKey } });
const validation = await client.validateProfessionalClaimSubmission(await loadClaim());
const errors = validation.errors ?? [];

if (errors.length === 0) {
  console.log("valid: the claim would be accepted for submission");
} else {
  console.log(`invalid: ${errors.length} error(s) would reject the claim:`);
  for (const error of errors) {
    console.log(`  ${error.description}`);
  }
}

File: examples/python/claims/validate_professional_claim_submission.py
# /// script
# requires-python = ">=3.12"
# dependencies = ["stedi==1.0.0"]
# ///

"""Check a professional claim without submitting it.

    STEDI_API_KEY=<api-key> uv run --script validate_professional_claim_submission.py

Runs the same payer checks, claim edits, and X12 mapping as
create_professional_claim_submission, but persists and delivers nothing — so you can
validate a claim while a user is still filling out a form. A response with no errors
means the claim would be accepted.
"""

import asyncio
import os
from pathlib import Path

from smithy_json import JSONCodec
from stedi import Config, Stedi
from stedi.models import ValidateProfessionalClaimSubmissionInput

FIXTURE = Path(__file__).parent / "fixtures" / "test-claim.json"


def load_claim() -> ValidateProfessionalClaimSubmissionInput:
    """Read the sample claim. Nothing is persisted, so no per-run stamping is needed."""
    claim = JSONCodec().deserialize(FIXTURE.read_bytes(), ValidateProfessionalClaimSubmissionInput)

    if not claim.payer or claim.payer.id != "STEDITEST":
        raise SystemExit("input must bill the Stedi test payer.")

    return claim


async def main(api_key: str) -> None:
    async with Stedi(Config(api_key=api_key)) as client:
        validation = await client.validate_professional_claim_submission(load_claim())

    errors = validation.errors or []
    if not errors:
        print("valid: the claim would be accepted for submission")
        return

    print(f"invalid: {len(errors)} error(s) would reject the claim:")
    for error in errors:
        print(f"  {error.description}")


api_key = os.environ.get("STEDI_API_KEY")
if not api_key:
    raise SystemExit("STEDI_API_KEY is not set")

asyncio.run(main(api_key))
